### ISSUE TITLE
Consider enabling breadcrumbs for more areas of the site

### ISSUE THREAD

**Issue by parlough** [dart-lang/site-www#5049]
### Describe the problem

Breadcrumbs are only enabled for linter rules currently. We can enable it for specific directories in the `_config.yml`.

Note however though it by default is generated from the path, so it only makes sense enabling it where the path represents cohesive areas of documentation and are following a structure we are happy with.

### Expected fix

Consider enabling it for more areas of the site, where it makes sense.

**Comment by MaryaBelanger** [comment]
> Note however though it by default is generated from the path,

So, for example, if we were to enable it for the Language pages _now_, the bread crumbs wouldn't represent the sidenav structure e.g. "Language > Types > Generics", it would just be "Language > Generics" because that's how the directory is organized?

This is like a way-down-the-line-once-we-have-a-standard-for-organization thought, but I wonder if having nested subdirectories under `/language` would be worth the more descriptive breadcrumbs... but then like you said, each one of those subdirectories would need an index, which would be a lot of bloat. Just thinking out loud, probably a UX question if anything!

**Comment by parlough** [comment]
> So, for example, if we were to enable it for the Language pages now, the bread crumbs wouldn't represent the sidenav structure e.g. "Language > Types > Generics", it would just be "Language > Generics" because that's how the directory is organized?

Yes exactly, at least with our current setup. Generally breadcrumb entries have to have a corresponding page.

> but I wonder if having nested subdirectories under /language would be worth the more descriptive breadcrumbs

That's what I thought we would have originally, just going with this since we weren't sure about the categories yet. The middle/index page, such as "Types", is not necessarily just an index, it could also be an overview/basic/intro page.

So for example, constructors could have "Language > Classes > Constructors" with a path of `/language/classes/constructors` and "Classes" could link to the existing "Classes" page at `/language/classes`.

On the Flutter site we just use plain indexes when no overview or intro page makes sense.
